### PR TITLE
Observable Item Add including Species Search

### PR DESCRIPTION
**Pull request by nspool**
Add an observable item create form containing an NRMN/WoRMS species lookup. Also use a query instead of the materialised view to provide data to the table of observable items.

Edit and View will redirect to a Work In Progress message for now.

http://nrmn-obs-item-add.dev.aodn.org.au

### CODE REVIEW

**Comment by nspool** [comment]
PR now ready for review.

**Comment by nspool** [comment]
Column order on Obserable Item Table appears to be wrong
Work In Progress is redirecting with a 401.

File: api/src/main/java/au/org/aodn/nrmn/restapi/controller/mapping/ObservableItemDtoMapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@Configuration
+public class ObservableItemDtoMapperConfig {
+
+    @Autowired
+    ObservableItemDtoMapperConfig(ObsItemTypeRepository obsItemTypeRepository, ModelMapper modelMapper) { 
+        Converter<Integer, ObsItemType> toObsItemType = ctx -> ctx.getSource() == null ? null :
+        obsItemTypeRepository.findById(ctx.getSource()).orElseThrow(SiteLocationNotFoundException::new);
+        modelMapper.typeMap(ObservableItemDto.class, ObservableItem.class).addMappings(mapper -> {
+            // mapper.using(toObsItemType).map(ObservableItemDto::getObsItemTypeId, ObservableItem::setObsItemType);
```
Comment (jonescc) [inline api/src/main/java/au/org/aodn/nrmn/restapi/controller/mapping/ObservableItemDtoMapperConfig.java:21]: TODO?

Comment (nspool) [reply api/src/main/java/au/org/aodn/nrmn/restapi/controller/mapping/ObservableItemDtoMapperConfig.java:21]: This is implemented in the obs item view / edit branch.

**Review by jonescc — APPROVED** [review]
LGTM - work still to be done but much better than what we have at the moment which is nothing.

**Pull request merged**